frontend/src/pages/offers/ViewOfferPage.tsx

```diff
@@ -32,26 +75,82 @@ const ViewOfferPage: FunctionComponent = () => {
           <h1 className="text-navy-800 text-3xl">Offer details</h1>
         </header>
         <main className="p-4 md:p-6 max-w-lg pb-20 space-y-6">
-          {data?.offer && (
+          {mutationError && (
+            <div className="p-4 rounded bg-red-50 mb-6 text-red-800">
+              <p className="font-semibold">Error:</p>
+              <p>{mutationError.message}</p>
+            </div>
+          )}
+          {offer?.offer && (
             <>
               <ReadOnlyField label="Shipment">
-                {targetShipment?.shipment && (
+                {shipment?.shipment && (
                   <>
-                    <p className="">
-                      {formatShipmentName(targetShipment.shipment)}
-                    </p>
+                    <p className="">{formatShipmentName(shipment.shipment)}</p>
                     <p className="text-gray-500 text-sm">
-                      {targetShipment.shipment.sendingHub.name} →{' '}
-                      {targetShipment.shipment.receivingHub.name}
+                      {shipment.shipment.sendingHub.name} →{' '}
+                      {shipment.shipment.receivingHub.name}
                     </p>
                   </>
                 )}
               </ReadOnlyField>
               <ReadOnlyField label="Group">
-                {data.offer.sendingGroupId}
+                {sendingGroup?.group.name}
               </ReadOnlyField>
+              <div className="flex justify-between items-center">
+                <h2 className="text-gray-700 font-semibold">Pallets</h2>
+                {offer.offer.pallets.length > 0 && (
+                  <Button
+                    disabled={mutationIsLoading}
+                    onClick={displayCreatePalletModal}
+                  >
+                    Add a pallet
+                  </Button>
+                )}
+              </div>
+              {offer.offer.pallets.length === 0 && (
+                <div className="p-4 flex flex-col space-y-4 items-center justify-center text-gray-500 bg-gray-50 rounded">
+                  <span>There are no pallets in this offer</span>
+                  <Button
+                    disabled={mutationIsLoading}
+                    onClick={displayCreatePalletModal}
+                  >
+                    Add a pallet
+                  </Button>
+                </div>
+              )}
+              {offer.offer.pallets.length > 0 && (
```

Comment:
I know there's a lot of markup in here but I'll move things around when I add the UI for creating line items, so I didn't wanna optimize too early.

Reply:
Totally fair